lib/src/parse/parser.dart

```diff
@@ -520,21 +520,26 @@ class Parser {
   /// Consumes the next character if it's equal to [letter], ignoring ASCII
   /// case.
   @protected
-  bool scanCharIgnoreCase(int letter) {
-    if (!equalsLetterIgnoreCase(letter, scanner.peekChar())) return false;
-    scanner.readChar();
-    return true;
+  bool scanAsciiChar(int letter, {bool caseSensitive = false}) {
+    var next = scanner.peekChar();
+    if (caseSensitive ? letter == next : equalsLetterIgnoreCase(letter, next)) {
```

Comment:
The relationship between this function and `StringScanner.scanChar()` would probably be clearer if you explicitly fell back to the latter when `caseSensitive` is `true`:

```dart
if (caseSensitive) return scanner.scanChar(letter);
if (!equalsLetterIgnoreCase(letter, scanner.peekChar())) return false;
scanner.readChar();
return true;
```